File: Infrastructure/Data/AuthService.cs
using System;
using Core.Interfaces;
using Core.Entities;
using Infrastructure.Services;

namespace Infrastructure.Data;

public class AuthService : IAuthService
{
    public LoginResponse? Login(LoginRequest request)
    {
        if (request.Username == "test" && request.Password == "1234")
        {
            var token = TokenService.GenerateToken(request.Username);
            return new LoginResponse
            {
                Token = token,
                Expiration = DateTime.UtcNow.AddHours(1)
            };
        }

        return null;
    }
}

File: client/src/app/core/services/auth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, tap } from 'rxjs';

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  expiration: string;
}

@Injectable({ providedIn: 'root' })
export class AuthService {
  private apiUrl = 'https://localhost:5001/auth/login';

  constructor(private http: HttpClient) {}

  login(data: LoginRequest): Observable<LoginResponse> {
    return this.http.post<LoginResponse>(this.apiUrl, data).pipe(
      tap(res => {
        localStorage.setItem('jwt_token', res.token); // token mentése
      })
    );
  }

  getToken(): string | null {
    return localStorage.getItem('jwt_token');
  }

  logout() {
    localStorage.removeItem('jwt_token');
  }
}
